[ feature request ] Add FreeBSD support

turso seems to require `io_uring`.

It would be great to have turso on FreeBSD, even if it was a bit slower without io_uring.

@penberg Thanks, here's the error:

```sh
error[E0425]: cannot find function `ptsname_r` in this scope                                                                                                                                                  
  --> /home/yonas/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rexpect-0.6.0/src/process.rs:97:26                                                                                                     
   |                                                                                                                                                                                                          
97 |         let slave_name = ptsname_r(&master_fd)?;                                                                                                                                                         
   |                          ^^^^^^^^^ not found in this scope                                                                                                                                               
                                                                                                                                                                                                              
error[E0432]: unresolved import `io::UnixIO`                                                                                                                                                                  
   --> core/lib.rs:57:9                                                                                                                                                                                       
    |                                                                                                                                                                                                         
57  | pub use io::UnixIO;                                                                                                                                                                                     
    |         ^^^^^^^^^^ no `UnixIO` in `io`                                                                                                                                                                  
    |                                                                                                                                                                                                         
note: found an item that was configured out                                                                                                                                                                   
   --> core/io/mod.rs:247:23                                                                                                                                                                                  
    |                                                                                                                                                                                                         
247 |         pub use unix::UnixIO;                                                                                                                                                                           
    |                       ^^^^^^                                                                                                                                                                            
note: the item is gated here                                                                                                                                                                                  
   --> core/io/mod.rs:240:1                                                                                                                                                                                   
    |                                                                                                                                                                                                         
240 | / cfg_block! {                                                                                                                                                                                          
241 | |     #[cfg(all(target_os = "linux", feature = "io_uring"))] {                                                                                                                                          
242 | |         mod io_uring;                                                                                                                                                                                 
243 | |         #[cfg(feature = "fs")]                                                                                                                                                                        
...   |                                                                                                                                                                                                       
279 | | }                                                                                                                                                                                                     
    | |_^                                                                                                                                                                                                     
note: found an item that was configured out                                                                                                                                                                   
   --> core/io/mod.rs:255:23                                                                                                                                                                                  
    |
255 |         pub use unix::UnixIO;
    |                       ^^^^^^
note: the item is gated here
   --> core/io/mod.rs:240:1
    |
240 | / cfg_block! {
241 | |     #[cfg(all(target_os = "linux", feature = "io_uring"))] {
242 | |         mod io_uring;
243 | |         #[cfg(feature = "fs")]
...   |
279 | | }
    | |_^
note: found an item that was configured out
   --> core/io/mod.rs:263:23
    |
263 |         pub use unix::UnixIO;
    |                       ^^^^^^
```

--- a/core/io/mod.rs
+++ b/core/io/mod.rs
@@ -330,14 +330,9 @@ cfg_block! {
         mod io_uring;
         #[cfg(feature = "fs")]
         pub use io_uring::UringIO;
-        mod unix;
-        #[cfg(feature = "fs")]
-        pub use unix::UnixIO;
-        pub use unix::UnixIO as SyscallIO;
-        pub use unix::UnixIO as PlatformIO;
     }
 
-    #[cfg(any(all(target_os = "linux",not(feature = "io_uring")), target_os = "macos"))] {
+    #[cfg(target_family = "unix")] {
         mod unix;
         #[cfg(feature = "fs")]
         pub use unix::UnixIO;

--- a/tests/Cargo.toml
+++ b/tests/Cargo.toml
@@ -32,7 +32,3 @@ zerocopy = "0.8.26"
 test-log = { version = "0.2.17", features = ["trace"] }
 tracing-subscriber = { version = "0.3.19", features = ["env-filter"] }
 tracing = "0.1.41"
-
-# rexpect does not support windows.
-[target.'cfg(not(windows))'.dependencies]
-rexpect = "0.6.0"

